File: src/main/java/org/dukcode/ps/codetree/trail2/lesson1/ChallengeRectangleWithANumber.java
package org.dukcode.ps.codetree.trail2.lesson1;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/challenge-rectangle-with-a-number/description">LINK</a>
 */
public class ChallengeRectangleWithANumber {

  private static BufferedReader br;
  private static BufferedWriter bw;

  private static int n;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    n = Integer.parseInt(br.readLine());

    printSquare(n);

    br.close();
    bw.close();

  }

  private static void printSquare(int n) throws IOException {
    for (int y = 0; y < n; y++) {
      for (int x = 0; x < n; x++) {
        int num = (y * n + x) % 9 + 1;
        bw.write(String.valueOf(num));
        bw.write(' ');
      }
      bw.newLine();
    }
    bw.newLine();
  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter01/lesson01/ChallengeRectangleWithANumber.kt
package org.dukcode.ps.codetree.trail02.chapter01.lesson01

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

/**
 * @see [LINK](https://www.codetree.ai/ko/trails/complete/curated-cards/challenge-rectangle-with-a-number/description)
 */
fun main() {
    val size = br.readLine().toInt()

    printSquare(size)

    bw.flush()
}

fun printSquare(size: Int) {
    var i = 0
    repeat(size) {
        repeat(size) {
            i = i % 9 + 1
            bw.write(i.toString())
            bw.write(' '.code)
        }
        bw.newLine()
    }
}
